worker_test.go

```diff
@@ -125,3 +135,76 @@ func startStopTest(celeryWorker *CeleryWorker, numWorkers int) error {
 	celeryWorker.StopWorker()
 	return nil
 }
+
+type retryableTask struct {
+	fail     bool
+	failFor  int
+	attempts int
+}
+
+func (r *retryableTask) Copy() (CeleryTask, error) { return r, nil }
+
+func (r *retryableTask) ParseKwargs(map[string]interface{}) error { return nil }
+
+func (r *retryableTask) RunTask() (interface{}, error) {
+	r.attempts++
+	if !r.fail {
+		return true, nil
+	}
+
+	if r.attempts <= r.failFor {
+		return nil, ErrTaskRetryable
+	}
+
+	return true, nil
+}
+
+func runRetryableTask(t *testing.T, c *CeleryClient, task Task, expectedAttempts int, shouldFail bool) {
+	res, err := c.Delay(task)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = res.Get(time.Second * 10)
+	if err != nil {
+		if shouldFail {
+			return
+		}
+
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	retryTask := c.worker.GetTask(task.Name).(*retryableTask)
+	if retryTask.attempts != expectedAttempts {
+		t.Fatalf("Attempts: Got=%v; Expected=%v", retryTask.attempts, expectedAttempts)
+	}
+}
+
+func checkRetryTask(t *testing.T, c *CeleryClient, taskName string) {
+	// dont fail
+	retryTask := &retryableTask{fail: false}
+	c.Register(taskName, retryTask)
+	runRetryableTask(t, c, Task{Name: taskName}, 1, false)
+
+	// fail for 2 times with 3 retries
+	retryTask = &retryableTask{fail: true, failFor: 2}
+	c.Register(taskName, retryTask)
+	runRetryableTask(t, c, Task{Name: taskName, Settings: DefaultSettings()}, 3, false)
+
+	// fail for 2 times with 3 retries
+	retryTask = &retryableTask{fail: true, failFor: 3}
+	c.Register(taskName, retryTask)
+	runRetryableTask(t, c, Task{Name: taskName, Settings: DefaultSettings()}, 2, true)
```

Comment:
here the number of attempts should be maxRetries + 1,
so here since the task numRetries = 3 the expectedAttemps should be = 4
as explained before,
`number of retries = number of tries(attempts) - 1`